### PR TITLE
[1.12.3] Signature verification: fix pull-by-digest

### PR DESCRIPTION
**Pull request by runcom**

### CODE REVIEW

**Comment by runcom** [comment]
Finally good to go, @rhatdan @mtrmac

**Comment by rhatdan** [comment]
LGTM, but Lets get @mtrmac review.

File: integration-cli/docker_cli_pull_local_test.go [hunk 1]
```diff
@@ -702,49 +702,40 @@ func (s *DockerRegistriesSuite) TestPullWithPolicy(c *check.C) {
 	dgst := matches[1]
 
 	// test that I can pull by a digest and tag when the signature is valid for the tag
```
Comment (mtrmac) [inline integration-cli/docker_cli_pull_local_test.go]: I can’t see a test for pulling by tag.

File: integration-cli/docker_cli_pull_local_test.go [hunk 2]
```diff
@@ -702,49 +702,40 @@ func (s *DockerRegistriesSuite) TestPullWithPolicy(c *check.C) {
...
-	_, _, err = dockerCmdWithError("pull", dgstRef)
-	c.Assert(err, check.NotNil)
+	dockerCmd(c, "pull", dgstRef)
 
 	dgstRefUnsigned := fmt.Sprintf("%s/busybox@%s", s.reg1.url, unsignedDigest)
 	out, _, err = dockerCmdWithError("pull", dgstRefUnsigned)
 	c.Assert(err, check.NotNil)
 	c.Assert(out, checker.Contains, "no signature exists")
 
 	// test that I can pull by tag but NOT by digest when the signature is valid for the tag
```
Comment (mtrmac) [inline integration-cli/docker_cli_pull_local_test.go]: I can’t see a test for pulling by tag.

Comment (runcom) [inline integration-cli/docker_cli_pull_local_test.go]: the first one in this chain

File: integration-cli/docker_cli_pull_local_test.go (hunk 1 above)
Comment (mtrmac) [inline integration-cli/docker_cli_pull_local_test.go]: So, `s/that I can pull by a digest and tag/thaat I can also pull by a digest/`?

File: integration-cli/docker_cli_pull_local_test.go (hunk 2 above)
Comment (mtrmac) [inline integration-cli/docker_cli_pull_local_test.go]: AFAICT there is only https://github.com/projectatomic/docker/pull/213/files#diff-8c285cfed55b08d027aed4e479cb9435R736 testing digest use.

**Comment by runcom** [comment]
@mtrmac fixed

> This is more or less a vendoring of containers/image, right? If so then that should be fine.

correct

**Comment by runcom** [comment]
@rhatdan good to go now!

**Pull request merged**